### PR TITLE
fix remote buildkit error: [415: Unsupported Media Type]

### PR DESCRIPTION
**Pull request by lizardruss**
# Issue describe
since legacy kube api not support server-side apply, when use remote buildkit will get the error:
```shell
local-registry: Starting Local Image Registry
build_images: build images: create builder: get or create local registry: ensure deployment: 415: Unsupported Media Type
fatal exit status 1
```
this pr will resolve the compatibility issue

# The pr summary
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**Please provide a short message that should be published in the DevSpace release notes**
fix remote buildkit error: [415: Unsupported Media Type], compatibility legacy kube api server which not support server-side apply

# reproduce
```shell
$ curl -Lo ~/bin/devspace-linux-amd64-6.3.12 https://github.com/devspace-sh/devspace/releases/download/v6.3.12/devspace-linux-amd64
$ chmod u+x ~/bin/devspace-linux-amd64-6.3.12

$ ~/bin/devspace-linux-amd64-6.3.12  version
DevSpace version : 6.3.12

$ kubectl version --short
Client Version: v1.26.7
Kustomize Version: v4.5.7
Server Version: v1.15.11

$ devspace-linux-amd64-6.3.12 build --skip-push
info Using namespace 'devspace'
info Using kube context 'devspace@dev'
Couldn't find a running docker daemon. Will fallback to local registry
local-registry: Starting Local Image Registry
build_images: build images: create builder: get or create local registry: ensure deployment: 415: Unsupported Media Type
fatal exit status 1
```

### CODE REVIEW

**Comment by lizardruss** [comment]
@wurenny This is a re-creation of your [original PR](https://github.com/devspace-sh/devspace/pull/2864) with some linting & test fixes.

File: pkg/devspace/kubectl/client.go [hunk 1]
```diff
@@ -17,7 +12,11 @@ import (
 	"github.com/loft-sh/devspace/pkg/util/log"
 	"github.com/loft-sh/devspace/pkg/util/survey"
 	"github.com/loft-sh/devspace/pkg/util/terminal"
+	"io"
 	"k8s.io/apimachinery/pkg/util/wait"
+	"net/http"
+	"os"
+	"time"
```
Comment (ThomasK33) [inline pkg/devspace/kubectl/client.go:19]: Those stdlib imports should be at the top, below fmt.

File: pkg/devspace/kubectl/util.go [hunk 2]
```diff
@@ -276,3 +276,13 @@ func GetKindContext(context string) string {
 
 	return strings.TrimPrefix(context, "kind-")
 }
+
+func IsIncompatibleServerError(err error) bool {
+	// 415: Unsupported media type means we're talking to a server which doesn't support server-side apply.
+	// Also included the apiserver enabled feature: ServerSideApply=false option
+	if _, ok := err.(*kerrors.StatusError); !ok {
```
Comment (ThomasK33) [inline pkg/devspace/kubectl/util.go:283]: We might want to use an [`error.As`](https://pkg.go.dev/errors#As) here.

Comment (lizardruss) [reply pkg/devspace/kubectl/util.go]: Just used `kerrors.IsUnsupportedMediaType(err)` instead.

**Pull request merged**